observation_portal/observations/viewsets.py

```diff
@@ -18,6 +18,17 @@
 from observation_portal.common.doc_examples import EXAMPLE_RESPONSES
 
 
+def get_sites_from_request(request):
```

Comment:
The changes in this file are all for the last scheduled timestamp updates for direct submissions.